src/main/webapp/app/entities/cc-member/member.tsx

```diff
@@ -0,0 +1,141 @@
+import { updateKey, updateKeys } from 'app/shared/util/deep-copy-utils';
+import { memberTabList } from 'app/shared/util/tab.constants';
+import axios from 'axios';
+import React, { useEffect, useState } from 'react';
+import { RouteComponentProps, useLocation } from 'react-router-dom';
+import { Dropdown, DropdownItem, DropdownMenu, DropdownToggle } from 'reactstrap';
+import CustomTab from 'app/shared/components/customTab/custom-tab';
+import { SearchEngine } from 'app/shared/util/native-search-utils';
+import { IUserCCInfo } from 'app/shared/model/user-cc-info.model';
+import { generateIndex } from './member.indexer';
+
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
+
+const member: React.FC<RouteComponentProps> = (props: RouteComponentProps) => {
+  const searchEngine: SearchEngine<IUserCCInfo> = new SearchEngine([], generateIndex);
+  const query = useQuery();
+  const [viewModel, setViewModel] = useState({
+    isLoading: false,
+    dropdownOpen: false,
+    selectedYearSession: '2019/2020',
+    memberName: '',
+    yearSessionList: ['2019/2020', '2020/2021', '2021/2022', '2022/2023', '2023/2024', '2024/2025', '2025/2026'],
```

Comment:
why this one hardcoded?
not get from BE?

Reply:
or this PR is just for UI?

Reply:
This file was pulled from #159. But for this PR, I mainly implement the search function into family-member search. Not sure why it still displaying this part of code as this is already existed in the dev-version-2 branch